cpp/include/raft/neighbors/sample_filter_types.hpp

```diff
@@ -37,6 +37,18 @@ struct none_ivf_sample_filter {
   }
 };
 
+/* A filter that filters nothing. This is the default behavior. */
```

Comment:
Can we please use one sample filtering API across all the ANN indexes? For example, a user should be able to instantiate a filter and use it in IVF-flat, ivf-pq, brute-force, or cagra. Is there anything stopping us from being able to do this?

Reply:
@lowener What do you think about the @cjnolet 's suggestion? This code is taken from your [PR](https://github.com/rapidsai/raft/pull/1765). I think we can use `none_ivf_sample_filter` as the default filter instead of `none_cagra_sample_filter`, but the name is not good because CAGRA is not ivf. I prefer to rename `none_ivf_sample_filter` to `none_sample_filter` but it changes the API so I'm not sure it is the best way for users.

Reply:
We have been communicating to users that sample filtering has been in an experimental state because we haven't had any actual optimized filtering primitives like bitsets, bitmaps, or bloom filters for them to use with the filters. It will be okay to make small adjustments to the APIs at this point, especially if they are consolidating allowing for more reuse.